src/chainstate/stacks/db/blocks.rs

```diff
@@ -5385,6 +5385,19 @@ impl StacksChainState {
                             .get_available_balance_at_burn_block(block_height),
                     ));
                 }
+
+                // if the payer for the tx is different from owner, check if they can afford fee
+                if origin != payer {
```

Comment:
Great catch on this!  Do we have a bunch of un-mineable transactions in the mempool that need to be cleared out?

Reply:
I looked at pending transactions in the Stacks explorer and the oldest pending transaction is from 3 days ago, so I *think* we should be fine on this?